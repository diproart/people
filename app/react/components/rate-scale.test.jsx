import React from 'react';
import RateScale from './rate-scale';
import {shallow, mount } from 'enzyme';

describe('RateScale', () => {

  beforeAll(() => {
    window.$ = function () {
      return {
        tooltip() {
          return 'test';
        }
      }
    }
  });

  describe('when rate type is range', () => {
    const testingComponent = <RateScale
      rate={0}
      rateType={'range'}
      onRateChange={fn => fn}
    />

    test('component structure match snapshot', () => {
      const subject = mount(testingComponent);
      expect(subject).toMatchSnapshot();
    });

    test('clicks on rate element change displayed elements', () => {
      const subject = shallow(testingComponent);
      expect(subject.find('.skill__clear_rate')).toHaveLength(0);
      expect(subject.find('.glyphicon-star-empty')).toHaveLength(3);
      expect(subject.find('.glyphicon-star')).toHaveLength(0);

      subject.find('li').children().first().simulate('click', { currentTarget: { dataset: { rate: 1 } } })

      expect(subject.find('.skill__clear_rate')).toHaveLength(1);
      expect(subject.find('.glyphicon-star-empty')).toHaveLength(2);
      expect(subject.find('.glyphicon-star')).toHaveLength(1);
    });

    test('hover on rate element toggle css classes', () => {
      const subject = shallow(testingComponent);

      expect(subject.find('.hovered')).toHaveLength(0);
      subject.find('li').children().first().simulate('mouseEnter', { currentTarget: { dataset: { rate: 3 } } })
      expect(subject.find('.hovered')).toHaveLength(3);
      subject.find('li').children().first().simulate('mouseLeave')
      expect(subject.find('.hovered')).toHaveLength(0);
    });

  });

  describe('when rate type is boolean', () => {
    const testingComponent = <RateScale
      rate={0}
      rateType={'boolean'}
      onRateChange={fn => fn}
    />

    test('component structure match snapshot', () => {
      const subject = mount(testingComponent);
      expect(subject).toMatchSnapshot();
    });

    test('clicks on rate element change displayed elements', () => {
      const subject = shallow(testingComponent);

      expect(subject.find('.skill__clear_rate')).toHaveLength(0);
      expect(subject.find('.glyphicon-unchecked')).toHaveLength(1);
      expect(subject.find('.glyphicon-check')).toHaveLength(0);

      subject.find('li').children().first().simulate('click', { currentTarget: { dataset: { rate: 1 } } })

      expect(subject.find('.skill__clear_rate')).toHaveLength(1);
      expect(subject.find('.glyphicon-unchecked')).toHaveLength(0);
      expect(subject.find('.glyphicon-check')).toHaveLength(1);
    });

    test('hover on rate element toggle css classes', () => {
      const subject = shallow(testingComponent);

      expect(subject.find('.hovered')).toHaveLength(0);
      subject.find('li').children().first().simulate('mouseEnter', { currentTarget: { dataset: { rate: 1 } } })
      expect(subject.find('.hovered')).toHaveLength(1);
      subject.find('li').children().first().simulate('mouseLeave')
      expect(subject.find('.hovered')).toHaveLength(0);
    });

  });
});
