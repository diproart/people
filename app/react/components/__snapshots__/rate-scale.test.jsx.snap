exports[`RateScale when rate type is boolean component structure match snapshot 1`] = `
<RateScale
  onRateChange={[Function]}
  rate={0}
  rateType="boolean">
  <ul
    className="list-inline skill__rating skill__rating--boolean">
    <li />
    <li>
      <i
        className="glyphicon skill__rate skill__rate--boolean glyphicon-unchecked"
        data-placement="top"
        data-rate={1}
        data-toggle="tooltip"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        title="skills.rating.boolean" />
    </li>
  </ul>
</RateScale>
`;

exports[`RateScale when rate type is range component structure match snapshot 1`] = `
<RateScale
  onRateChange={[Function]}
  rate={0}
  rateType="range">
  <ul
    className="list-inline skill__rating skill__rating--range">
    <li />
    <li>
      <i
        className="glyphicon skill__rate skill__rate--range glyphicon-star-empty"
        data-placement="top"
        data-rate={1}
        data-toggle="tooltip"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        title="skills.rating.range" />
    </li>
    <li>
      <i
        className="glyphicon skill__rate skill__rate--range glyphicon-star-empty"
        data-placement="top"
        data-rate={2}
        data-toggle="tooltip"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]} />
    </li>
    <li>
      <i
        className="glyphicon skill__rate skill__rate--range glyphicon-star-empty"
        data-placement="top"
        data-rate={3}
        data-toggle="tooltip"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]} />
    </li>
  </ul>
</RateScale>
`;
